crates/client/src/https_client_connection.rs

```diff
@@ -25,11 +24,11 @@ pub struct HttpsClientConnection<T> {
     name_server: SocketAddr,
     dns_name: String,
     client_config: ClientConfig,
-    marker: PhantomData<T>,
+    runtime: T,
 }
 
-impl<T> HttpsClientConnection<T> {
-    /// Creates a new client connection.
+impl<T: RuntimeProvider + Default> HttpsClientConnection<T> {
```

Comment:
I'm almost in favor of removing this, but maybe we should just deprecate it instead?